### PR TITLE
Add derive `PartialEq` to `DiscoveryResult`

### LINKED ISSUE #175: Add `std::cmp::PartialEq` trait to `DiscoveryResult` to facilitate unit testing
**Is your feature request related to a problem? Please describe.**

`DiscoveryResult` does not permit equality comparisons which would be useful in e.g. unit testing.

The struct is defined:

```rust
#[derive(Debug, Clone)]
pub struct DiscoveryResult {
    pub digest: String,
    pub properties: HashMap<String, String>,
}
```

**Is your feature request related to a way you would like Akri extended? Please describe.**

Adding `PartialEq` permits e.g. `assert_eq!(_: DiscoveryResult, _: DiscoveryResult)` and this facilitates unit testing.

**Describe the solution you'd like**

```rust
#[derive(Debug, Clone, PartialEq)]
pub struct DiscoveryResult {
    pub digest: String,
    pub properties: HashMap<String, String>,
}
```

### PR DESCRIPTION
**Pull request by DazWilkin**
Fixes #175

Promotes unit testing by permitting e.g. `assert_eq!(_: DiscoverResult, _: DiscoveryResult)`

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [X] added code adheres to standard Rust formatting (`cargo fmt`)
- [X] code builds properly (`cargo build`)
- [X] code is free of common mistakes (`cargo clippy`)
- [X] all Akri tests succeed (`cargo test`)
- [ ] inline documentation builds (`cargo doc`)
- [ ] version has been updated appropriately (`./version.sh`)

### CODE REVIEW

File: agent/src/protocols/mod.rs [hunk 1]
```diff
@@ -8,7 +8,7 @@ use blake2::VarBlake2b;
 use failure::Error;
 use std::collections::HashMap;
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq)]
```
Comment (bfjelds) [inline agent/src/protocols/mod.rs:11]: maybe add a unit test to validate:

```rust
    #[tokio::test]
    async fn test_discovery_result_partialeq() {
        let left = DiscoveryResult::new(&"foo1".to_string(), HashMap::new(), true);
        let right = DiscoveryResult::new(&"foo1".to_string(), HashMap::new(), true);
        assert_eq!(left, right);
   }

    #[tokio::test]
    async fn test_discovery_result_partialeq_false() {
        {
            let left = DiscoveryResult::new(&"foo1".to_string(), HashMap::new(), true);
            let right = DiscoveryResult::new(&"foo2".to_string(), HashMap::new(), true);
            assert_ne!(left, right);
        }

        {
            let left = DiscoveryResult::new(&"foo1".to_string(), HashMap::new(), true);
            let right = DiscoveryResult::new(&"foo1".to_string(), HashMap::new(), false);
            assert_ne!(left, right);
        }

        {
            let mut nonempty = HashMap::new();
            nonempty.insert("one", "two");
            let left = DiscoveryResult::new(&"foo1".to_string(), nonempty, true);
            let right = DiscoveryResult::new(&"foo1".to_string(), HashMap::new(), true);
            assert_ne!(left, right);
        }
   }
```

Comment (DazWilkin) [reply agent/src/protocols/mod.rs:11]: I may need some guidance on your environment variable mocker (see line 186)

Fixed: https://github.com/deislabs/akri/pull/176/commits/63c4bbbe9e690036e012c66dbacb727efa290c89

Comment (bfjelds) [reply agent/src/protocols/mod.rs:11]: interesting ... i didn't see that coming, lol.  

to do this right, we would need to essentially modify `DiscoveryResult::new` to accept `query: &impl EnvVarQuery` as a parameter (like `inner_get_discovery_handler`), then use query in `DiscoveryResult::new`.  in the test, we'd use a `MockEnvVarQuery` like `test_factory_for_debug_echo_when_no_env_var_set()`.

but that might be more of a change than you are interested in making :)

another possibility would be to remove that `DisoceryResult::new(..., false)` test (or comment it out with a note saying some work needs to be done to `DiscoveryResult::new` to enable the test)

Comment (DazWilkin) [reply agent/src/protocols/mod.rs:11]: If ok with you, I'll exclude the `false` shared for now with a comment and then, perhaps next week (?), I'll learn what you've explained and submit a follow-up PR?

Comment (DazWilkin) [reply agent/src/protocols/mod.rs:11]: Per discussion: https://github.com/deislabs/akri/pull/176/commits/1b02e672b49851fde7d6bce26f310021839b5193

**Comment by kate-goldenring** [comment]
@DazWilkin, can you bump the patch version? `./version.sh -u -p`?

**Review by bfjelds — APPROVED** [review]

**Comment by DazWilkin** [comment]
> @DazWilkin, can you bump the patch version twice with `./version.sh -u -p`? We want it 0.0.1 greater than https://github.com/deislabs/akri/blob/main/version.txt

Fixed: https://github.com/deislabs/akri/pull/176/commits/6859c3f43bffe65e1a38363b9a4273f5fb871310

**Comment by kate-goldenring** [comment]
> > @DazWilkin, can you bump the patch version twice with `./version.sh -u -p`? We want it 0.0.1 greater than https://github.com/deislabs/akri/blob/main/version.txt
> 
> Fixed: [6859c3f](https://github.com/deislabs/akri/commit/6859c3f43bffe65e1a38363b9a4273f5fb871310)

Sorry, I wasn't super clear. You'll need to do `./version.sh -u -p` one more time to make the version greater than main. Currently they match. We should update the version.sh script to check what the current version in main is instead of making this a manual process

**Review by kate-goldenring — APPROVED** [review]

**Review by bfjelds — APPROVED** [review]

**Pull request merged**